src/services/subscription-deferred-install-service.ts

```diff
@@ -0,0 +1,66 @@
+import { Request } from "express";
+import IORedis from "ioredis";
+import { getRedisInfo } from "config/redis-info";
+import { v4 as newUUID } from "uuid";
+
+export interface SubscriptionDeferredInstallPayload {
+	installationIdPk: number,
+	gitHubServerAppIdPk?: number,
+	gitHubInstallationId: number,
+	orgName: string
+}
+
+const redis = new IORedis(getRedisInfo("SubscriptionDeferredInstallService"));
+
+const REDIS_CLEANUP_TIMEOUT = 2 * 24 * 3600 * 1000;
+
+const toRedisKey = (requestId: string) => "sub_def_inst_serv_req}" + requestId;
+
+const toRequestId = (req: Request) => req.params["requestId"];
+
+export const extractSubscriptionDeferredInstallPayload = async (req: Request) => {
```

Comment:
**IMPORTANT**
I don't think we should make the `service` aware of http stuff, such as `Request` here. Would prefer a typed param.